tempesta_fw/hpack.c

```diff
@@ -1834,7 +1834,7 @@ do {									\
 			/* All duplicate headers must be skipped by caller. */
 			WARN_ON_ONCE(test_bit(i, mit->found));
 			__set_bit(i, mit->found);
-			if (!TFW_STR_CHUNK(dc_iter->desc->hdr, 2)) {
+			if (!TFW_STR_CHUNK(dc_iter->desc->hdr, 1)) {
```

Comment:
There is comment `All duplicate headers must be skipped by caller.` at the above - please update it with the words why do we check the 2nd chunk, i.e. why strings with only one chunks are treated as duplicate headers.

Reply:
The comment about duplicates is applied only for two following lines, while `if (!TFW_STR_CHUNK(dc_iter->desc->hdr, 1))...` has nothing to do about duplicates. I agree, that is looks misleading and messed up, so I've added a new comment here, describing chunk indexes.